plugins/foreman_tasks/0.8/index.md

```diff
@@ -0,0 +1,302 @@
+            :after: 10d
+
+The foreman_tasks:cleanup script also accepts additional parameters to specify the search criteria for the cleanup manually:
+
+- **TASK_SEARCH** - scoped search filter (example: 'label = "Actions::Foreman::Host::ImportFacts"')
+- **AFTER** - delete tasks created after AFTER period. Expected format is a number followed by the time unit (s, h, m, y), such as 10d for 10 days (applicable only when the TASK_SEARCH option is specified)
+- **STATES** - comma separated list of task states to touch with the cleanup, by default only stopped tasks are affected (applicable only when the TASK_SEARCH option is specified)
+- **NOOP** - set to "true" if the task should not actually perform the deletion, only report the actions the script would perform
+- **VERBOSE** - set to "true" for more verbose output
+- **BATCH_SIZE** - the size of batches the tasks get processed in (1000 by default)
+
+To see the current configuration (what actions get cleaned automatically and what is their after period), this script can be used:
+
+    foreman-rake foreman_tasks:cleanup:config
+
+## 3.8 Exporting
+
+If you want to share the the history of tasks with someone else who does not have access to your Foreman instance, there's a tool to do the export. This is especially useful when developers asks you for debugging some debugging data. To do the export, use to following rake task provided by the plugin
+
+    foreman-rake foreman_tasks:export_tasks
+
+This will create gzipped tar file in temp directory containing html page for each exported task. The path to the file is printed after the export is finished. These data contains all tasks there are present in database. To limit what exact tasks should be part of the export, use following variables when running the task.
+
+- **TASK_SEARCH** - scoped search filter (example: 'label = "Actions::Foreman::Host::ImportFacts"')
+- **TASK_DAYS** - number of days to go back
+
+To explicitly name the export, you can use **TASK_FILE** variable. The following example would export all facts importing tasks from last week into file /tmp/last_week_importing.tar.gz
+
+    foreman-rake foreman_tasks:export_tasks TASK_SEARCH='label = "Actions::Foreman::Host::ImportFacts"' \
+      TASK_FILE=/tmp/last_week_importing.tar.gz TASK_DAYS=7
+
+If you prefer CSV format instead of HTML pages, you use use **TASK_FORMAT** variable and change to format to csv. Only html and csv formats are available at the moment. The CSV format does not generate the gzipped tarball but plain CSV file that contains limited information of matched tasks. The information includes task id, state, type, label, result, parent_task_id, started_at, ended_at attributes.
+
+
+# 4. Developer guide
+
+We don't have the developer guide yet. At the moment we recommend to read the code in the [repository](https://github.com/theforeman/foreman-taskshttps://github.com/theforeman/foreman-tasks). Please consider contributing.
```

Comment:
Thanks, fixed. I think ActiveJob way should be mentioned here but I think there are still cases where you want to use advanced features of foreman-tasks, such as locking, that requires you to write the task manually. So I think the developer guide still makes sense covering stuff like locking, humanized_output etc. Anyway, I'd appreciate if you put together examples with ActiveJob API to this manual once it's ready.